pkg/controller/appdefinition/testdata/deployspec/karpenter/input.yaml

```diff
@@ -0,0 +1,31 @@
+kind: AppInstance
+apiVersion: internal.acorn.io/v1
+metadata:
+  name: app-name
+  namespace: app-namespace
+  uid: 1234567890abcdef
+spec:
+  image: test
+status:
+  namespace: app-created-namespace
+  appImage:
+    id: test
+  appSpec:
+    containers:
+      scalenil:
```

Comment:
New test added to make sure we are setting the proper annotation when acorn has different scale.